.github/workflows/build.yml

```
@@ -0,0 +1,36 @@
+name: Build and Run TypeScript project
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install Dependencies
+        run: npm install
```

```suggestion
        run: npm ci
``` think this is the better command for pipeline